commands.py

```diff
@@ -96,32 +96,58 @@ def run(self, _: sublime.Edit) -> None:
 class CopilotAcceptSuggestionCommand(CopilotTextCommand):
     @_provide_session()
     def run(self, session: Session, edit: sublime.Edit) -> None:
-        completion = Completion(self.view)
-        if not completion.is_visible:
+        completion_manager = ViewCompletionManager(self.view)
+        if not completion_manager.is_visible:
             return
 
-        completion.hide()
+        completion_manager.hide()
+
+        completion = completion_manager.current_completion
+        if not completion:
+            return
 
         # Remove the current line and then insert full text.
         # We don't have to care whether it's an inline completion or not.
-        source_line_region = self.view.line(completion.region[1])
+        source_line_region = self.view.line(completion["positionSt"])
         self.view.erase(edit, source_line_region)
-        self.view.insert(edit, source_line_region.begin(), completion.text)
+        self.view.insert(edit, source_line_region.begin(), completion["text"])
+
+        # notify the current completion as accepted
+        self._record_telemetry(session, REQ_NOTIFY_ACCEPTED, {"uuid": completion["uuid"]})
 
-        # TODO: When a suggestion is accept, we need to send a REQ_NOTIFY_REJECTED
-        # request with all other completions which weren't accepted
-        self._record_telemetry(session, REQ_NOTIFY_ACCEPTED, {"uuid": completion.uuid})
+        # notify all other completions as rejected
+        other_uuids = [completion["uuid"] for completion in completion_manager.completions]
+        other_uuids.remove(completion["uuid"])
+        if other_uuids:
+            self._record_telemetry(session, REQ_NOTIFY_REJECTED, {"uuids": other_uuids})
 
 
 class CopilotRejectSuggestionCommand(CopilotTextCommand):
     @_provide_session()
     def run(self, session: Session, _: sublime.Edit) -> None:
-        completion = Completion(self.view)
-        completion.hide()
+        completion_manager = ViewCompletionManager(self.view)
+        completion_manager.hide()
+
+        # notify all completions as rejected
+        self._record_telemetry(
+            session,
+            REQ_NOTIFY_REJECTED,
+            {"uuids": [completion["uuid"] for completion in completion_manager.completions]},
```

Comment:
Why do we report all `uuid`s? Say, I see a suggestion and I not happy with it, why other suggestions should be also marked as reject as I might have not even seen them?

Reply:
@TheSecEng since this is his comment TODO.

Reply:
I'll analyze VSCode copilot again, but from my brief understand, they record which suggestions where selected as a ways to determine more context on your codebase. 

So when you accept one, it implies the others weren't accepted. 

I'll double check the code again but I believe this is their intent

Reply:
Let's merge this PR. If telemetry parts need to be changed, we can do it in another PR later.